lucille-core/src/test/java/com/kmwllc/lucille/indexer/ElasticsearchIndexerTest.java

```diff
@@ -416,6 +420,149 @@ public void testBulkResponseErroring() throws Exception {
     }
   }
 
+  /**
+   * Tests that the indexer correctly ignores fields stated in the ignoreFields portion of the config file
+   * Note that indexer would even ignore the "id" field if configured, removing the id field in the Lucille document.
+   * However, the id would still be passed to the ElasticSearch index to id the documents.
+   * @throws Exception
+   */
+  @Test
+  public void testIgnoreFields() throws Exception {
+    TestMessenger messenger = new TestMessenger();
+    Config config = ConfigFactory.load("ElasticsearchIndexerTest/ignoreFields.conf");
+
+    Document doc = Document.create("doc1");
+    doc.setField("ignoreField1", "value1");
+    doc.setField("ignoreField2", "value2");
+    doc.setField("normalField", "normalValue");
+
+    // checking that document fields has been set
+    assertNotNull(doc.getString("id"));
+    assertNotNull(doc.getString("ignoreField1"));
+    assertNotNull(doc.getString("ignoreField2"));
+    assertNotNull(doc.getString("normalField"));
```

Comment:
Since setField is (hopefully!) being tested elsewhere in the codebase, I don't think its necessary to assert these fields exist after you set them.

If you agree we don't need these just remember to remove them for all the tests.

Reply:
yes those were tested in DocumentTest. Will remove those.